framework_tvb/tvb/core/services/import_service.py

```diff
@@ -219,80 +219,102 @@ def _store_imported_images(self, project, temp_project_path, project_name):
                 if metadata_file.endswith(FilesHelper.TVB_FILE_EXTENSION):
                     self._import_image(root, metadata_file, project.id, target_images_path)
 
-    def import_project_operations(self, project, import_path):
-        """
-        This method scans provided folder and identify all operations that needs to be imported
-        """
-        imported_operations = []
-
+    def get_directory_ordered_list(self, project, import_path):
+        directory_list = {}
         for root, _, files in os.walk(import_path):
-            # TODO This LOOP will import operations in a random order.
-            # We should order by Op (in case of XML) Create Date or ViewModel.create_date
             if "Operation.xml" in files:
-                # Previous Operation format for uploading previous versions of projects
                 operation_file_path = os.path.join(root, "Operation.xml")
                 operation = self.__build_operation_from_file(project, operation_file_path)
-                operation.import_file = operation_file_path
-                self.logger.debug("Importing operation " + str(operation))
-                operation_entity, datatype_group = self.__import_operation(operation)
-                new_op_folder = self.files_helper.get_project_folder(project, str(operation_entity.id))
-                # TODO ViewModel H5 should be created, as we want to preserve only the new structure in the future
-                operation_datatypes = self._load_datatypes_from_operation_folder(root, operation_entity,
-                                                                                 datatype_group, new_op_folder)
-                self._store_imported_datatypes_in_db(project, operation_datatypes)
-                imported_operations.append(operation_entity)
-
+                directory_list[operation.create_date] = root
```

Comment:
How about when 2 operations are created at the same time ?
Theoretically, are small chances, but it could happen in case of PSE, for example .